src/components/loginRedirect/LoginRedirect.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { BackHandler, Image, Platform, View } from 'react-native'
+import web3 from 'web3'
+
+import { CustomButton, Section, Text, Wrapper } from '../common'
+import useProfile from '../../lib/userStorage/useProfile'
+import API from '../../lib/API/api'
+import goodWallet from '../../lib/wallet/GoodWallet'
+
+export const LoginRedirect = ({
+  navigation: {
+    state: { params },
+  },
+}) => {
+  const { email, walletAddress, mobile, fullName } = useProfile()
+  const [details, setDetails] = useState({ country: '', isWhitelisted: null })
+  const [urlDetails, setURLDetails] = useState({
+    vendorAddress: null,
+    vendorName: null,
+    vendorURL: null,
+    urlType: null,
+    url: null,
+  })
+  const [warning, setWarnings] = useState({
+    isVendorWalletWhitelisted: true,
+    isWebDomainDifferent: false,
+  })
+  const { isVendorWalletWhitelisted, isWebDomainDifferent } = warning
+  const { country, isWhitelisted } = details
+
+  const fetchLocationAndWhiteListedStatus = useCallback(async () => {
+    try {
+      const urlDetails = JSON.parse(Buffer.from(decodeURIComponent(params.login), 'base64').toString('ascii'))
+      setURLDetails({
+        vendorAddress: urlDetails.id,
+        vendorName: urlDetails.v,
+        vendorURL: urlDetails.web,
+        urlType: urlDetails.cbu ? 'cbu' : 'rdu',
+        url: urlDetails?.cbu ?? urlDetails?.rdu,
+      })
+      if (!urlDetails.web.includes(urlDetails?.cbu ?? urlDetails?.rdu)) {
+        setWarnings(data => ({ ...data, isWebDomainDifferent: true }))
+      }
+      if (!(await goodWallet.isVerified(urlDetails.id))) {
+        setWarnings(data => ({ ...data, isVendorWalletWhitelisted: false }))
+      }
+      const location = await API.getLocation()
+      const isWhitelisted = await goodWallet.isCitizen()
+      setDetails({ country: location?.name, isWhitelisted })
+    } catch (e) {
+      //nothing here
+    }
+  }, [])
+
+  useEffect(() => {
+    fetchLocationAndWhiteListedStatus()
+  }, [fetchLocationAndWhiteListedStatus])
+
+  const submit = ({ isRejected = false }) => {
+    const web3Instance = new web3()
+    let responseurlDetails = {}
+    if (!isRejected) {
+      responseurlDetails = {
+        a: walletAddress,
+        v: isWhitelisted,
+        I: country,
+        n: fullName,
+        e: email,
+        m: mobile,
+        nonce: Date.now(),
+      }
+      responseurlDetails.sig = web3Instance.eth.accounts.sign(responseurlDetails, goodWallet.accounts[0].privateKey)
+      delete responseurlDetails.sig.message
+    } else {
+      responseurlDetails = {
+        error: 'Authorization Denied',
+      }
+    }
+    if (Platform.OS === 'ios' || Platform.OS === 'android') {
```

Comment:
we have helpers for that, search code for isMobileNative

Reply:
done